### PR TITLE
Running SAM backbone on frontend

### LINKED ISSUE #5984: Feature Idea: Incorporate "Segment Anything"
Hello, it is great that you support out-of-the-box models like YoloV7, do you also plan to include the latest FAI model "Segment-Anything"? I think that could be very helpful!

https://github.com/facebookresearch/segment-anything

Kind regards

### LINKED ISSUE #6041: Gateway Timeout (504) error when running with SAM
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Steps to Reproduce (for bugs)

1. Downloaded `cvat`, am on commit `ad534b2ac32f57`.
2. Installed NVIDIA container toolkit.
1. Followed [Serverless Setup](https://opencv.github.io/cvat/docs/manual/advanced/serverless-tutorial/#optimize-using-gpu) steps.
1. Installed `nuctl` by following guide [here](https://opencv.github.io/cvat/docs/administration/advanced/installation_automatic_annotation/). Verified that `nuclio` is version `1.8.14`.
1. Ran command to launch SAM nuctl function as described [here](https://www.cvat.ai/post/facebook-segment-anything-model-in-cvat). `cd serverless && ./deploy_gpu.sh pytorch/facebookresearch/sam/nuclio/`

1. Checked that `nuclio` function is running properly `nuctl get function` returns that SAM function is in STATE `ready`.
1. Launched CVAT in serverless mode using `docker-compose -f docker-compose.yml -f components/serverless/docker-compose.serverless.yml up -d`.
1. Open CVAT task, select "Segment Anything" from AI tools, click on image. Get a "Waiting a response from Segment Anything." After a while I get a 504 timeout error. `Failed to load resource: the server responded with a status of 504 (Gateway Timeout)`. Clicking on the link in the browser console shows me REST api call (image below).
3. 
image

### Current Behaviour
It seems that CVAT instance is unable to communicate with the nuclio SAM function. I have verified that SAM function is running in nuclio dashboard.

### Your Environment

- Git hash commit `ad534b2a`:
- Docker version `23.0.4`
- Are you using Docker Swarm or Kubernetes? Regular docker
- Operating System and version (e.g. Linux, Windows, MacOS):. Linux 22.04

### LINKED ISSUE #6049: Better Integration for Neural Embedding based workflowes (eg: SAM (Segment Anything))
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

On large images, SAM latency per query adds up.

### Possible Solution
Is there a way we can store the encoded image embedding in CVAT db, perhaps a separate table and then run the interactive prompt encoder + decoder with onnx.js?

The table would have a primary key on the `project/task/job/frame` with columns for different model embeddings. We can use a simple binary blob to store the data. This table can be populated by either by task (workflow already exists) or per frame if the db doesn't have an entry.

This would require adding support for arbitrary scripts/webasm binaries in the browser to enable use of onnx.js

### Context
Scaling DB is much easier (and cheaper) than scaling GPU HW, specially since the GPU compute is idempotent

### Current Implementation and Thoughts
1. Create a table manually in the database: Easy, but official support would be appreciated
2. Create a detector model which gets the request and populates the table, and doesn't detect anything ever
    * Feels like a hack to have 2 models, one for pre-process and one for use
    * Need to access poorly documented event.body payload
3. Create an interactor which runs pytorch for the prompt encoder and decoder as a serverless function
    * Sadly, making this use onnx.js would take too much effort (and know-how of cvat and js)
    * We add a checksum to the db to ensure the image is the same, else generate embeddings again

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Resolved #5984 
Resolved #6049
Resolved #6041

- Compatible only with ``sam_vit_h_4b8939.pth`` weights. Need to re-export ONNX mask decoder with some custom model changes (see below) to support other weights (or just download them using links below)
- Need to redeploy the serverless function because its interface has been changed.

Decoders for other weights:
sam_vit_l_0b3195.pth: [Download](https://drive.google.com/file/d/1Nb5CJKQm_6s1n3xLSZYso6VNgljjfR-6/view?usp=sharing)
sam_vit_b_01ec64.pth: [Download](https://drive.google.com/file/d/17cZAXBPaOABS170c9bcj9PdQsMziiBHw/view?usp=sharing)

Changes done in ONNX part:
```
git diff scripts/export_onnx_model.py
diff --git a/scripts/export_onnx_model.py b/scripts/export_onnx_model.py
index 8441258..18d5be7 100644
--- a/scripts/export_onnx_model.py
+++ b/scripts/export_onnx_model.py
@@ -138,7 +138,7 @@ def run_export(

     _ = onnx_model(**dummy_inputs)

-    output_names = ["masks", "iou_predictions", "low_res_masks"]
+    output_names = ["masks", "iou_predictions", "low_res_masks", "xtl", "ytl", "xbr", "ybr"]

     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", category=torch.jit.TracerWarning)
bsekachev@DESKTOP-OTBLK26:~/sam$ git diff segment_anything/utils/onnx.py
diff --git a/segment_anything/utils/onnx.py b/segment_anything/utils/onnx.py
index 3196bdf..85729c1 100644
--- a/segment_anything/utils/onnx.py
+++ b/segment_anything/utils/onnx.py
@@ -87,7 +87,15 @@ class SamOnnxModel(nn.Module):
         orig_im_size = orig_im_size.to(torch.int64)
         h, w = orig_im_size[0], orig_im_size[1]
         masks = F.interpolate(masks, size=(h, w), mode="bilinear", align_corners=False)
-        return masks
+        masks = torch.gt(masks, 0).to(torch.uint8)
+        nonzero = torch.nonzero(masks)
+        xindices = nonzero[:, 3:4]
+        yindices = nonzero[:, 2:3]
+        ytl = torch.min(yindices).to(torch.int64)
+        ybr = torch.max(yindices).to(torch.int64)
+        xtl = torch.min(xindices).to(torch.int64)
+        xbr = torch.max(xindices).to(torch.int64)
+        return masks[:, :, ytl:ybr + 1, xtl:xbr + 1], xtl, ytl, xbr, ybr

     def select_masks(
         self, masks: torch.Tensor, iou_preds: torch.Tensor, num_points: int
@@ -132,7 +140,7 @@ class SamOnnxModel(nn.Module):
         if self.return_single_mask:
             masks, scores = self.select_masks(masks, scores, point_coords.shape[1])

-        upscaled_masks = self.mask_postprocessing(masks, orig_im_size)
+        upscaled_masks, xtl, ytl, xbr, ybr = self.mask_postprocessing(masks, orig_im_size)

         if self.return_extra_metrics:
             stability_scores = calculate_stability_score(
@@ -141,4 +149,4 @@ class SamOnnxModel(nn.Module):
             areas = (upscaled_masks > self.model.mask_threshold).sum(-1).sum(-1)
             return upscaled_masks, scores, stability_scores, areas, masks

-        return upscaled_masks, scores, masks
+        return upscaled_masks, scores, masks, xtl, ytl, xbr, ybr
```

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx [hunk 1]
```diff
@@ -867,25 +882,70 @@ export class ToolsControlComponent extends React.PureComponent<Props, State> {
...
 
+    private async initializeOpenCV(): Promise<void> {
+        if (!openCVWrapper.isInitialized) {
+            const hide = message.loading('OpenCV client initialization..', 0);
+            try {
+                await openCVWrapper.initialize(() => {});
+            } finally {
+                hide();
+            }
+        }
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx]: Done

File: cvat-ui/plugins/sam_plugin/src/ts/index.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    description: string;
+    cvat: {
+        lambda: {
+            call: {
+                enter: (
+                    plugin: SAMPlugin,
+                    taskID: number,
+                    model: any,
+                    args: any,
+                ) => Promise<null | { preventMethodCall: boolean }>;
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply cvat-ui/plugins/sam_plugin/src/ts/index.tsx]: Done

**Comment by d710055071** [comment]
Does this implement operations related to point prompt, box prompt, point+box prompt, and mask
